scipy/optimize/tests/test_hessian_update_strategy.py

```diff
@@ -30,38 +31,61 @@ def grad(self, x):
         xm_m1 = x[:-2]
         xm_p1 = x[2:]
         der = np.zeros_like(x)
-        der[1:-1] = (200 * (xm - xm_m1**2) -
-                     400 * (xm_p1 - xm**2) * xm - 2 * (1 - xm))
-        der[0] = -400 * x[0] * (x[1] - x[0]**2) - 2 * (1 - x[0])
-        der[-1] = 200 * (x[-1] - x[-2]**2)
+        der[1:-1] = (200 * (xm - xm_m1 ** 2) -
+                     400 * (xm_p1 - xm ** 2) * xm - 2 * (1 - xm))
+        der[0] = -400 * x[0] * (x[1] - x[0] ** 2) - 2 * (1 - x[0])
+        der[-1] = 200 * (x[-1] - x[-2] ** 2)
         return der
 
     def hess(self, x):
         x = np.atleast_1d(x)
         H = np.diag(-400 * x[:-1], 1) - np.diag(400 * x[:-1], -1)
         diagonal = np.zeros(len(x), dtype=x.dtype)
-        diagonal[0] = 1200 * x[0]**2 - 400 * x[1] + 2
+        diagonal[0] = 1200 * x[0] ** 2 - 400 * x[1] + 2
         diagonal[-1] = 200
-        diagonal[1:-1] = 202 + 1200 * x[1:-1]**2 - 400 * x[2:]
+        diagonal[1:-1] = 202 + 1200 * x[1:-1] ** 2 - 400 * x[2:]
         H = H + np.diag(diagonal)
         return H
 
 
 class TestHessianUpdateStrategy(TestCase):
 
     def test_hessian_initialization(self):
-        quasi_newton = (BFGS(), SR1())
 
-        for qn in quasi_newton:
-            qn.initialize(5, 'hess')
-            B = qn.get_matrix()
+        ndims = 5
+        symmetric_matrix = np.array([[43, 24, 33, 34, 49],
+                                     [24, 36, 44, 15, 44],
+                                     [33, 44, 37, 1, 30],
+                                     [34, 15, 1, 5, 46],
+                                     [49, 44, 30, 46, 22]])
+        init_scales = (
+            ('auto', np.eye(ndims)),
+            (2, np.eye(ndims) * 2),
+            (np.arange(1, ndims + 1), np.arange(1, ndims + 1) * np.eye(ndims)),
+            (symmetric_matrix, symmetric_matrix),)
+        for init_scale, true_matrix in init_scales:
+            # large min_{denominator,curvatur} makes them skip an update, so we can have our initial matrix
```

Comment:
I meant in terms of character length here even though we don't track PEP8 in test files there is no need to have extra violations.

Reply:
Ah sure, I've adjusted that